app/logic/Mail/OWA/Request/OWACreateItemRequest.ts

```diff
@@ -1,15 +1,13 @@
-export default class OWACreateItemRequest {
-  readonly __type = "CreateItemJsonRequest:#Exchange";
-  readonly Header = {
-    __type: "JsonRequestHeaders:#Exchange",
-    RequestServerVersion: "Exchange2013",
-  };
+import OWARequest from "./OWARequest";
+
+export default class OWACreateItemRequest extends OWARequest {
   Body: any = {
     __type: "CreateItemRequest:#Exchange",
     Items: [{}],
   };
 
-  constructor(attributes?: {[key: string]: string | boolean | object}) {
+  constructor(attributes?: { [key: string]: string | boolean | object }) {
+    super("CreateItemJsonRequest");
```

Comment:
It's not obvious so I thought I would mention that the `super` constructor actually tries to set `Body` according to the second parameter, but as it happens the assignment to `Body` from the class definition occurs immediately after the `super` constructor returns, so everything is great. (Not so great for the various calendar event classes, which want to do similar things to different bodies, sigh...)

Reply:
If we can avoid such "smartness", that would be good. More indirection is a) much harder to read and understand the code and b) more difficult to refactor, if a requirement violates the assumption made here.